napari/layers/utils/text_manager.py

```diff
@@ -51,92 +49,97 @@ class TextManager(EventedModel):
         Offset from the anchor point.
     rotation : float
         Angle of the text elements around the anchor point. Default value is 0.
+    color : ColorEncoding
+        Defines the color for each text element.
+    text : StringEncoding
+        Defines the string for each text element.
+
+    Private Attributes
+    ------------------
+    _n_text : int
+        The number of text elements managed, which should correspond to the number
+        of rows in the properties table. Should be removed if/when the properties
+        table reliably stores that count.
     """
 
-    values: Array[str] = []
+    # Declare properties as a generic dict so that a copy is not made on validation
```

Comment:
Nice. If a user tries to overwrite this will they get an error? Should it be private?

Reply:
Ideally this would be private. Unfortunately, we need access to this field when validating `text` and `color` - currently to check if the `str` passed in refers to a property name (i.e. as an ergonomic shortcut), and maybe later for error checking.

I'm not aware of a way to do that in Pydantic because private attributes are not fields, aren't validated, and aren't available in `values` of validators that come after them.

We could make this field final (i.e. set `allow_mutation=False`), but then the owning layer would have to recreate `TextManager` when its `properties` changes. I think that's fine because setting properties will trigger a full update/refresh anyway, so I'll probably pursue this, but also open to other ideas.

Reply:
makes sense, this is fine